opds.py

```diff
@@ -747,6 +747,26 @@ def make_link(ep):
             cached.update(_db, content)
         return content
 
+    @classmethod
+    def from_query(cls, query, _db, library, lane, list, url, pagination, url_fn, annotator):
+        """Build  a feed representing one page of a given list. Currently used for
+        creating an OPDS feed for a custom list and not cached.
+        """
+        page_of_works = pagination.apply(query)
+
+        feed = cls(_db, list.name, url, page_of_works, annotator)
+
+        if len(list.entries) > 0 and pagination.has_next_page:
```

Comment:
`Pagination.has_next_page` only works if `.total_size` or `.this_page_size` has been set on the Pagination object. It's your responsibility to set those values -- otherwise `has_next_page` will always return True.

Something like this should work:

```
page_of_works = pagination.apply(query).all()
pagination.this_page_size = len(page_of_works)
```

Reply:
+1, I noticed this bug when I was testing this.

Could you do `pagination.apply(query).count()` though?

Reply:
Thanks, `pagination.total_size = query.count()` seems to give me the more consistent result that's needed.